### PR TITLE
[RFR] Refactoring authentication routes

### PR DESCRIPTION
**Pull request by alexisjanvier**

### CODE REVIEW

File: src/api/authentication/authenticateApiRoutes.js [hunk 1]
```diff
@@ -16,9 +16,25 @@ app.use(function* init(next) {
...
+
+    this.body = {
+        id: user.id,
+        email: user.email,
+        token: jwt.sign(user, config.apps.api.security.jwt.privateKey),
+    };
+}));
 
-app.use(koaRoute.post('/sign-up', function* () {
+app.use(koaRoute.post('/sign-up', function* signIn() {
```
Comment (djhi) [inline src/api/authentication/authenticateApiRoutes.js]: signUp ;)

File: src/api/authentication/authenticateApiRoutes.js [hunk 2]
```diff
@@ -16,9 +16,25 @@ app.use(function* init(next) {
     yield next;
 });
 
-app.use(koaMount('/authenticate', authenticateRoutes));
+app.use(koaRoute.post('/authenticate', rateLimiter(config.apps.api.security.rateLimitOptions)));
```
Comment (djhi) [inline src/api/authentication/authenticateApiRoutes.js]: as discussed, this should probably be renamed `sign-in`

File: src/api/authentication/authenticateAdminRoutes.js [hunk 3]
```diff
@@ -1,9 +1,32 @@
...
 const app = koa();
 
-app.use(koaMount('/authenticate', authenticateRoutes));
+app.use(methodFilter(['POST']));
+app.use(koaRoute.post('/', rateLimiter(config.apps.api.security.rateLimitOptions)));
+
+app.use(koaRoute.post('/', function* login() {
+    const { email, password } = yield coBody(this);
+    const userRepository = userRepositoryFactory(this.client);
+    const user = yield userRepository.authenticate(email, password);
```
Comment (djhi) [inline src/api/authentication/authenticateAdminRoutes.js]: inside a `try catch` with the catch:
```js
catch(err) {
    this.throw(err);
}
```
?

Comment (djhi) [inline src/api/authentication/authenticateAdminRoutes.js]: You really should use `this.throw(401)` or do as suggested in previous comment

Comment (alexisjanvier) [inline src/api/authentication/authenticateAdminRoutes.js]: Yes, sure :) It's just that i didn't see your comment when i done the review.

Comment (alexisjanvier) [inline src/api/authentication/authenticateAdminRoutes.js]: Now, `userRepository.authenticate` return a `user` false if user is not authenticate. It's the controller which will throw an error (and set the status code to 401).

**Comment by Kmaschta** [comment]
Question: koa return an http response if an error is threw ?

**Comment by alexisjanvier** [comment]
IMO Yes : `koa uses http-errors to create errors.`
https://github.com/koajs/koa/blob/master/docs/api/context.md#ctxthrowmsg-status-properties

**Comment by alexisjanvier** [comment]
@djhi The review is applied

**Comment by djhi** [comment]
:+1:

**Pull request merged**